### PR TITLE
add interface of get amop subscriber

### PR DESCRIPTION
**Pull request by MaggieNgWu**

### CODE REVIEW

**Review by cyjseagull — APPROVED** [review]

File: librpc/Rpc.cpp [hunk 1]
```diff
@@ -553,6 +553,67 @@ Json::Value Rpc::getGroupList()
...
+        Json::Value nodes = Json::Value(Json::arrayValue);
+        auto sessions = service()->sessionInfos();
+        
+        for (auto it = sessions.begin(); it != sessions.end(); ++it)
+        {
+            for (auto topic : it->topics)
+            {
+                if (topic.topic.compare(_topicName)==0 && topic.topicStatus == dev::TopicStatus::VERIFYI_SUCCESS_STATUS)
+                {
+                    nodes.append(boost::lexical_cast<std::string>(it->nodeIPEndpoint));
```
Comment (bxq2011hust) [inline librpc/Rpc.cpp:575]: also return nodeID

File: librpc/Rpc.cpp [hunk 2]
```diff
@@ -553,6 +553,67 @@ Json::Value Rpc::getGroupList()
...
+
+        // search node subscribers
+        Json::Value nodes = Json::Value(Json::arrayValue);
+        auto sessions = service()->sessionInfos();
+        
+        for (auto it = sessions.begin(); it != sessions.end(); ++it)
+        {
+            for (auto topic : it->topics)
+            {
+                if (topic.topic.compare(_topicName)==0 && topic.topicStatus == dev::TopicStatus::VERIFYI_SUCCESS_STATUS)
```
Comment (bxq2011hust) [inline librpc/Rpc.cpp:573]:
```suggestion
                if (topic.topic.compare(_topicName) == 0 && topic.topicStatus == dev::TopicStatus::VERIFYI_SUCCESS_STATUS)
```

File: librpc/Rpc.cpp [hunk 3]
```diff
@@ -553,6 +553,67 @@ Json::Value Rpc::getGroupList()
...
+        }
+        
+
+        // search sdk subscribers
+        Json::Value sdks = Json::Value(Json::arrayValue);
+        auto channelRPCServer = m_ledgerInitializer->channelRPCServer();
+        std::vector<dev::channel::ChannelSession::Ptr>  activedSessions = channelRPCServer->getSessionByTopic(_topicName);
+        for (auto& it : activedSessions)
+        {   
+            sdks.append(it->host()+":"+std::to_string(it->port()));
```
Comment (bxq2011hust) [inline librpc/Rpc.cpp:588]: also return sdk pubkey and agency

Comment (bxq2011hust) [inline librpc/Rpc.cpp:588]:
```suggestion
            sdks.append(it->host() + ":" + std::to_string(it->port()));
```

File: librpc/Rpc.cpp [hunk 4]
```diff
@@ -553,6 +553,67 @@ Json::Value Rpc::getGroupList()
...
+        int size =0;
+        if (!nodes.isNull()){
+            size += nodes.size();
+        }
+        if (!sdks.isNull()){
+            size += sdks.size();
+        }
+    
+        response["total"]=size;
+        response["nodes"]=nodes;
```
Comment (bxq2011hust) [inline librpc/Rpc.cpp:602]:
```suggestion
        response["nodes"] = nodes;
```

File: librpc/Rpc.cpp (hunk 1 above)
Comment (MaggieNgWu) [reply librpc/Rpc.cpp:576]: OK

**Pull request merged**